tensorflow/python/ops/array_grad.py

```diff
@@ -354,9 +354,15 @@ def _PreventGradientGrad(op, _):
 def _GatherGrad(op, grad):
   """Gradient for Gather op."""
   # params can be large, so colocate the shape calculation with it.
+  #
+  # params can be super-large for sparse model, array_ops.shape raises
+  # exception inside Windows when any dimension is larger than int32,
```

Comment:
Nits:
s/super-large/very large/
s/inside Windows/on the Windows platform/

Period after int32.

Maybe the last sentence should be

"params_shape is not used in apply_sparse gradients, so it's fine to convert it back to int32 regardless of truncation."